source/extensions/access_loggers/filters/local_ratelimit/filter.cc

```diff
@@ -0,0 +1,41 @@
+#include "source/extensions/access_loggers/filters/local_ratelimit/filter.h"
+
+#include "envoy/access_log/access_log.h"
+
+#include "source/common/init/target_impl.h"
+#include "source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace AccessLoggers {
+namespace Filters {
+namespace LocalRateLimit {
+LocalRateLimitFilter::LocalRateLimitFilter(
+    Server::Configuration::FactoryContext& context,
+    const envoy::extensions::access_loggers::filters::local_ratelimit::v3::LocalRateLimitFilter&
+        config) {
+  rate_limiter_ = Envoy::Extensions::Filters::Common::LocalRateLimit::RateLimiterProviderSingleton::
+      getRateLimiter(
+          context, config.resource_name(), config.config_source(),
+          [this](std::shared_ptr<
```

Comment:
This can be deleted when the callback fires. You should pass a weak_ptr<shared_from_this()>. The limiter could also be passed as reference, to avoid copying shared pointers.

Reply:
How about now?